docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+
+
+A Chart is composed of the following pieces:
+
+
+1. A human- and machine-readable metadata file called `Chart.yaml`
+2. A directory named `templates/`
+3. An optional `README.md` file
+4. An optional `LICENSE` file
+5. An optional `docs/` directory
+6. An optional `hooks/` directory
+7. An optional `icon.svg.`
+
+
+A chart _may_ have other directories and files, but such files are not defined by or required by this proposal. For the purposes of this proposal, any additional files do not contribute to the core functionality of chart installation.
+
+
+_Note: When a chart file is deployed, a [provenance file](#the-provenance-file) is generated for the chart. That file is not stored inside of the chart, but is considered part of the chart’s packaged format._
+
+
+The chart.yaml file format is described [later in this document](#the-chart-file).
+
+
+The `templates/` directory contains one or more template files, as defined in the [Directory Layout](#directory-layout) section. Templates, along with all of their supporting files, are located therein.
+
+
+An optional `README.md` file may be specified, which contains long-form text information about using this chart. Tools may display this information, if present. The README.md file is in Markdown format, and should contain information about a Chart’s purpose, usage, and development.
+
+
+An optional `LICENSE` file may be present, which specifies license information for this chart and/or the images dependent on it.
+
+
+An optional `docs/` directory may be present, and may contain one or more documentation files. This directory contains documentation that is more specific, verbose, or thorough than what is present in the `README.md` file.
+
+
+An optional `hooks/` directory may be present. **This is reserved for future use.** A deployment engine may specify lifecycle hooks, such as `pre-install`, `pre-render`, `post-install` and `post-render`. Lifecycle hooks provide an opportunity for a chart author to run additional steps in the processing pipeline. Implementations of these hooks are stored here. _No other files may be stored in this directory._
+
+### Directory Layout
+A chart is laid out as follows. The top level directory (represented by the placeholder ROOT) must be the name of the chart. For example, if the chart is named `nginx`, the ROOT directory must be named `nginx/`.
+
+```
+ROOT/
+    Chart.yaml
+    README.md
+    LICENSE
+    docs/
+        some.md
+    templates/
+        some.yaml
+        some.jinja
+        some.jinja.schema
+```
+
+Templates are stored in a separate directory for the following reasons:
+
+
+* This future-proofs the format: we can add other directories at the top level and deprecate `templates/`.
+* It allows authors to add other files (such as documentation) in a straightforward way that will not confuse definitions-aware tools.
+* It allows for the possibility that a definition may be embedded inside of project code.
```

Comment:
Seems like the idea is to not litter the base dir with an arbitrary number of files.  The name "templates" may conflict but that's the best you can do.  On the other hand it's not future-proof because you can't add more directories at the top level in a backwards compatible way.

A less common name than "templates" might be a good idea as I wouldn't be surprised if lots of repos already have a directory called templates.